### PR TITLE
Move away from `cached_download`

### PR DESCRIPTION
**Pull request by Wauplin**
Follow-up PR after https://github.com/huggingface/diffusers/pull/8417.

In #8417, we introduce a mirror of the community pipeline modules as a HF dataset. This means we can now download modules from HF using `hf_hub_download` directly.

### CODE REVIEW

File: src/diffusers/utils/dynamic_modules_utils.py [hunk 1]
```diff
@@ -282,16 +277,15 @@ def get_cached_module_file(
             )
 
         # community pipeline on GitHub
-        github_url = COMMUNITY_PIPELINES_URL.format(revision=revision, pipeline=pretrained_model_name_or_path)
         try:
-            resolved_module_file = cached_download(
-                github_url,
+            resolved_module_file = hf_hub_download(
+                repo_id="diffusers/community-pipelines-mirror",
```
Comment (sayakpaul) [inline src/diffusers/utils/dynamic_modules_utils.py:282]: (nit): `repo_id` could be a constant here.

**Review by sayakpaul — COMMENTED** [review]
This design looks good to me!

**Comment by Wauplin** [comment]
@sayakpaul I've addressed your comment + added a custom error to ask to open an issue if a revision is not correctly pushed. Hopefully it'll be fine for 99.9% of users using `main` or a recent-enough tag. Let me know what you think.

**Comment by sayakpaul** [comment]
Killer work here!

**Pull request merged**